Генерация дымовых тестов по отобранным объектам метаданных

Kyrales opened the pull request:
Ишузы:
https://github.com/Pr-Mex/vanessa-automation/issues/2246
https://github.com/Pr-Mex/vanessa-automation/issues/2378

Генерация дымовых тестов по отобранным объектам метаданных

- Интерактивная отметка опций в дереве метаданных, с необходимым набором функций (открыть форму объекта, запись и др.)
- Генерация сценариев дымовых тестов в виде единого текста
- Возможность исключать из дерева метаданных ранее уже существующие дымовые тесты 
- Возможность исключать из дерева метаданных только введённые объекты (Справочники, Документы, Планы видов характеристик)
- Совместимость с механизмом пакетного формирования фича-файлов
- Создание/обновление фича-файлов дымовых тестов 
- Команда "Подготовка данных" - при нажатии открывает форму "Подготовки и генерации данных", с проставлением отметки ранее выбранным в дереве ссылочных объектов метаданных
- Исправлены ошибки https://github.com/Pr-Mex/vanessa-automation/issues/2415

<img width="1664" height="934" alt="image" src="https://github.com/user-attachments/assets/7d5a9d40-bb36-4917-a5c9-90deb19e98b2" />

В планах в будущих ПР:

- Дополнительные условия для отборов "Расширенные действия" (Запись, Копия). Гибкое изменение условий для запросов, чтобы не нужно было дополнительно работать с консолью запросов.

Kyrales committed:
Генерация дымовых тестов по отобранным объектам метаданных

- Интерактивная отметка в дереве метаданных с необходимым набором опций
- Генерация сценария в коде
- Возможность исключать из дерева метаданных ранее уже существующие дымовые тесты
- Возможность исключать из дерева метаданных только введённые объекты (Справочники, Документы, Планы видов характеристик)
- Совместимость с механизмом пакетного формирования фича-файлов

Pr-Mex commented:
run tests

Pr-Mex commented:
@Kyrales 
Упало несколько тестов. Посмотри пожалуйста
http://vanessa.bit-erp.ru/job/VAPullRequestCheck/903/allure/#behaviors/6a2b14b5401f09a5fb51867fae7d5e1d/721f1bcf5768aa64/

Kyrales committed:
fix: Исправлены ошибки по тестам и дополнены сценарии тестирования

Kyrales commented:
> @Kyrales Упало несколько тестов. Посмотри пожалуйста http://vanessa.bit-erp.ru/job/VAPullRequestCheck/903/allure/#behaviors/6a2b14b5401f09a5fb51867fae7d5e1d/721f1bcf5768aa64/

@Pr-Mex ошибки исправил

Kyrales commented:
run tests

Pr-Mex commented:
run tests

Kyrales commented:
@Pr-Mex что по результатам?

Kyrales committed:
feat: Добавлено гибкое управление сформированными сценариями по метаданным и обновление feature-файлов (для новых сценариев)

1. Добавлена колонка "Есть записи", которая устанавливается если в объектах метаданных (Справочники, Документы, ПВХ), есть хотя бы одна запись
2. Добавлена функции анализа и обновления файлов-сценариев из "Каталог feature-файлов" (фича-файлы должны быть созданы заранее на основе "Пакетное формирование фича-файлов")
3. Добавлена команда "Подготовка данных" - при нажатии открывает форму "Подготовки и генерации данных", с проставлением отметки ранее выбранным в дереве ссылочных объектов метаданных
4. Добавлен признак "Показать полный текст кода сценария 1С", для отображение полного исходного кода всех сценариев в одном текстовом документе
5. Обновлены тесты

Kyrales commented:
@Pr-Mex  дополнен функционал
feat: Добавлено гибкое управление сформированными сценариями по метаданным и обновление feature-файлов (для новых сценариев)

1. Добавлена колонка "Есть записи", которая устанавливается если в объектах метаданных (Справочники, Документы, ПВХ), есть хотя бы одна запись
2. Добавлена функции анализа и обновления файлов-сценариев из "Каталог feature-файлов" (фича-файлы должны быть созданы заранее на основе "Пакетное формирование фича-файлов")
3. Добавлена команда "Подготовка данных" - при нажатии открывает форму "Подготовки и генерации данных", с проставлением отметки ранее выбранным в дереве ссылочных объектов метаданных
4. Добавлен признак "Показать полный текст кода сценария 1С", для отображение полного исходного кода всех сценариев в одном текстовом документе
5. Обновлены тесты

<img width="1650" height="921" alt="image" src="https://github.com/user-attachments/assets/6719b45e-421f-486c-9ebe-cdd59d05a66e" />

Pr-Mex commented:
run tests

Pr-Mex commented:
@Kyrales 
Привет. Я нажимаю "Обновить feature-файлы", но файлы в каталоге не появляются.
Так и должно быть?
<img width="1662" height="934" alt="image" src="https://github.com/user-attachments/assets/ccdc894d-01ec-4217-b76d-b15b5d5ba3fc" />

Kyrales commented:
> @Kyrales Привет. Я нажимаю "Обновить feature-файлы", но файлы в каталоге не появляются. Так и должно быть? 

@Pr-Mex Привет. Да все верно, так и должно быть. В папке этой уже должны быть фича-файлы, которые ранее выгружены через базовый функционал на закладке "Пакетное формирование фича-файлов". Тут специально не дублируем функционал по созданию файлов. 

Алгоритм использования:
1. При первичном использовании, когда нужны новые дымовые тесты (и их раньше не создавали) мы используем по старинке ""Пакетное формирование фича-файлов"  по типам объектов метаданных
2. Далее в процессе доработок конфигурации/расширений появляются новые объекты метаданных и нужно обновлять фича-файлы созданные из п.1 - вот тут и нужно будет пользоваться этой командой "Обновить feature-файлы" на закладке "Формирование сценариев по метаданным"

Pr-Mex commented:
@Kyrales 
На мой взгляд это не очевидно.
Новая закладка "Формирование сценариев по метаданным" выглядит как замена предыдущей (поправь, плиз, если это не так).
А если это замена - то пользователю было бы удобно сразу всё формировать в одном окне.

Pr-Mex commented:
@Kyrales 
И тогда кнопку можно было бы переименовать в "Обновить/создать feature-файлы"

Kyrales commented:
> @Kyrales На мой взгляд это не очевидно. Новая закладка "Формирование сценариев по метаданным" выглядит как замена предыдущей (поправь, плиз, если это не так). А если это замена - то пользователю было бы удобно сразу всё формировать в одном окне.

@Pr-Mex По части работы с feature-файлами, это было запланировано не как замена, а именно дополнение. Пользователь если хочет может сформировать по метаданным исходный текст и потом куда хочет его скопировать. Или он может действующие ранее сформированные feature-файлы обновить.

Я в целом могу дополнить вот таким признаком, чтобы еще и создавались feature-файлы:
<img width="1650" height="921" alt="image" src="https://github.com/user-attachments/assets/019678ed-eac7-4865-bb43-58613e398d6d" />

Pr-Mex commented:
@Kyrales 
Думаю можно не добавлять новый флаг, а делать такое поведение по умолчанию.

Pr-Mex commented:
@Kyrales 
Или при нажатии на кнопку задать пользователю вопрос:
"Нет feature файлов для обновления. Создать их?" (что-то в этом стиле)

Kyrales committed:
feat: Добавлен функционал создания feature-файлов

+ Исправлены ошибки оформления
+ Обновлены тесты

Kyrales commented:
@Pr-Mex  
Добавил функционал создания feature-файлов. 
При нажатии на команду "Создать/обновить feature-файлы": Окно с вопросом будет появляться, только в том случае если будут сценарии без фича-файлов.
Плюс еще поправил оформление кода и обновил тесты.

<img width="985" height="858" alt="image" src="https://github.com/user-attachments/assets/a196ac2f-5c08-4650-9e7b-426db8231b75" />
<img width="1664" height="934" alt="image" src="https://github.com/user-attachments/assets/e50f294d-738f-4cbf-8186-eb8d66cddbee" />

Pr-Mex commented:
run tests

Kyrales committed:
fix: Исправлены ошибки вывода формы объекта в дымовых (была ранее) и ошибка формирования печатных форм в новых БСП

Kyrales commented:
@Pr-Mex 
Исправлены две ошибки из https://github.com/Pr-Mex/vanessa-automation/issues/2415 , т.к. мешали тестированию

Pr-Mex commented:
run tests

Kyrales committed:
fix: Ошибка в тесте "Проверка генерации дымовых тестов из командной строки и корректного закрытия сеанса" в "Каталоге выходных файлов"

Kyrales commented:
@Pr-Mex перезапусти плз тесты. Исправил сейчас ошибку по сценарию "Проверка генерации дымовых тестов из командной строки и корректного закрытия сеанса"

Pr-Mex commented:
run tests

Pr-Mex commented:
@Kyrales 
Сейчас не могу остановить предыдущий пайп. Придётся ждать его завершения.

Kyrales commented:
@Pr-Mex я хочу создать еще один фича файл FeatureGeneration_SSL.feature. Хочу использовать для тестов в БСП для проверки дымовых тестов. Надо чтобы этот фича-файл не брался в проверки на твоих CI-CD, т.к. тесты чисто под демо-БСП базу будут. Какие теги выставить в этом фича-файле, чтобы не подхватывался в CI-CD?

<img width="1281" height="1032" alt="image" src="https://github.com/user-attachments/assets/4a166382-842a-43b6-94c6-08e5a5a158ea" />

Pr-Mex commented:
@Kyrales 
Надо добавить тег @IgnoreOnCIMainBuild

Kyrales commented:
@Pr-Mex после последнего анализа, дефектов по своей доработке Генератора дымовых тестов не вижу.
Что это за ошибки?

<img width="1039" height="680" alt="image" src="https://github.com/user-attachments/assets/4fad02f0-4b51-4e73-9c5f-64c6dc239a2f" />

Pr-Mex commented:
@Kyrales 
Это не ошибки реквеста. Я проверял локально.
С ними буду отдельно разбираться.

Ты закончил с реквестом? Проверяю-принимаю?

Kyrales commented:
> @Kyrales Это не ошибки реквеста. Я проверял локально. С ними буду отдельно разбираться.
> 
> Ты закончил с реквестом? Проверяю-принимаю?

@Pr-Mex да закончил. И так он раздулся. Если уже что-то буду добавлять, то отдельными ПР

Pr-Mex commented:
run tests

Pr-Mex commented:
@Kyrales 
Я залил свои изменения и теперь все тесты проходят.
Я постараюсь в выходные добраться до реквеста.

Pr-Mex commented:
@Kyrales 
Привет!
Посмотрел. В целом всё ок.
Можешь описать изменения по этому реквесту тут?
https://github.com/Pr-Mex/vanessa-automation/blob/develop/docs/Smoke/Smoke.md

Kyrales committed:
docs: Обновлена документация

Kyrales commented:
> @Kyrales Привет! Посмотрел. В целом всё ок. Можешь описать изменения по этому реквесту тут? https://github.com/Pr-Mex/vanessa-automation/blob/develop/docs/Smoke/Smoke.md

@Pr-Mex Привет. Обновил описание

Kyrales committed:
docs: Обновлена документация. Корректировка

Pr-Mex commented:
@Kyrales 
Спасибо!
